pkg/apis/monitoring/v1/types.go

```diff
@@ -2022,6 +2022,9 @@ type AlertmanagerSpec struct {
 	ForceEnableClusterMode bool `json:"forceEnableClusterMode,omitempty"`
 	// AlertmanagerConfigs to be selected for to merge and configure Alertmanager with.
 	AlertmanagerConfigSelector *metav1.LabelSelector `json:"alertmanagerConfigSelector,omitempty"`
+	// The AlertmanagerConfigMatcherStrategy defines how to add additional matchers to the AlertmanagerConfigs.
+	// In the future more options may be added.
+	AlertmanagerConfigMatcherStrategy *AlertmanagerConfigMatcherStrategy `json:"alertmanagerConfigMatcherStrategy,omitempty"`
```

Comment:
We can avoid a pointer here.

```suggestion
	AlertmanagerConfigMatcherStrategy AlertmanagerConfigMatcherStrategy `json:"alertmanagerConfigMatcherStrategy,omitempty"`
```

Reply:
My point was to say that if we use a value rather than a pointer, we don't need to check for nil pointers (as [here](https://github.com/prometheus-operator/prometheus-operator/blob/fa2ca180fe4a7fac2e9204622b6b9de98beb0914/pkg/alertmanager/amcfg.go#L199)).

Reply:
Ah I see, sure my bad.